src/System.Text.JsonLab.Dynamic/System/Text/Json/JsonDynamicObject.cs

```diff
@@ -218,6 +291,67 @@ private static unsafe bool TryEncodeControlChar(SymbolTable symbolTable, byte va
             return symbolTable.TryEncode(value, buffer, out written);
         }
 
+        private static int GetHashCode(ReadOnlySpan<byte> span)
+        {
+            int hash = 17;
+            foreach (byte element in span)
+            {
+                hash = hash * 31 + element;
+            }
+            return hash;
+        }
+
+        private static Dictionary<int, PropertyInfoLinkedList> GetTypeMap(Type type)
+        {
+            var dictionary = new Dictionary<int, PropertyInfoLinkedList>();
+            IEnumerable<PropertyInfo> properties = type.GetRuntimeProperties();
+            foreach (PropertyInfo pi in properties)
```

Comment:
Would it make sense for this map to be lazy? i.e. you add new propoerties on demand.

Reply:
This is a one time operation (per type), so the cost is amortized over multiple calls to Deserialize. Regardless, we could make it lazy if we generally expect the json payload to partially fill the object.